core/src/network/reducers/publish.rs

```diff
@@ -23,25 +19,17 @@ fn publish_entry(
 ) -> Result<(), HolochainError> {
     let entry_with_header = util::EntryWithHeader::from((entry.clone(), header.clone()));
 
-    let data = DhtData {
-        msg_id: "?".to_string(),
-        dna_hash: network_state.dna_hash.clone().unwrap(),
-        agent_id: network_state.agent_id.clone().unwrap(),
-        address: entry.address().to_string(),
-        content: serde_json::from_str(&serde_json::to_string(&entry_with_header).unwrap()).unwrap(),
-    };
-
-    network_state
-        .network
-        .as_mut()
-        .map(|network| {
-            network
-                .lock()
-                .unwrap()
-                .send(ProtocolWrapper::PublishDht(data).into())
-                .map_err(|error| HolochainError::IoError(error.to_string()))
-        })
-        .expect("Network has to be Some because of check above")
+    send(
+        network_state,
+        ProtocolWrapper::PublishDht(DhtData {
+            msg_id: "?".to_string(),
+            dna_hash: network_state.dna_hash.clone().unwrap(),
+            agent_id: network_state.agent_id.clone().unwrap(),
+            address: entry.address().to_string(),
+            content: serde_json::from_str(&serde_json::to_string(&entry_with_header).unwrap())
```

Comment:
can't we use JsonString here?

Reply:
We can and we should. But I didn’t want to change the net crates in this PR and instead just use it. Follow up.